zebra/zapi_msg.c

```diff
@@ -2755,9 +2755,9 @@ static inline void zread_iptable(ZAPI_HANDLER_ARGS)
 		zebra_pbr_add_iptable(zpi);
 	else
 		zebra_pbr_del_iptable(zpi);
-
+	return;
 stream_failure:
-	zebra_pbr_iptable_free(zpi);
+	XFREE(MTYPE_TMP, zpi);
```

Comment:
Why did you change this? This is an obvious leak of the list allocated on 2735.

Reply:
the list should be updated when one adds/Remove an iptable entry.

Reply:
The list allocated here is never saved. If I'm wrong, please post the code where that happens.

Reply:
https://github.com/FRRouting/frr/blob/master/zebra/zebra_pbr.c#L704

Reply:
Yeah, I'm not seeing it.
https://github.com/FRRouting/frr/blob/master/zebra/zebra_pbr.c#L685
New struct is allocated, list duplicated, all elements copied.